chain/jsonrpc-primitives/src/types/blocks.rs

```diff
@@ -0,0 +1,99 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(thiserror::Error, Debug)]
+pub enum RpcBlockError {
+    #[error("Block `{0}` is missing")]
+    BlockMissing(near_primitives::hash::CryptoHash),
+    #[error("Block not found")]
+    BlockNotFound(String),
+    #[error("There are no fully synchronized blocks yet")]
+    NotSyncedYet,
+    #[error("The node reached its limits. Try again later. More details: {0}")]
+    InternalError(String),
+    #[error("It is a bug if you receive this error type, please, report this incident: https://github.com/near/nearcore/issues/new/choose. Details: {0}")]
+    Unreachable(String),
```

Comment:
Sorry for getting here once again, but let's also clarify why it is a bug, and why it should be unreachable.

```suggestion
    // NOTE: Currently, the underlying errors are too broad, and while we tried to handle
    // expected cases, we cannot statically guarantee that no other errors will be returned
    // in the future.
    // TODO #3851: Remove this variant once we can exhaustively match all the underlying errors
    #[error("It is a bug if you receive this error type, please, report this incident: https://github.com/near/nearcore/issues/new/choose. Details: {0}")]
    Unreachable(String),
```